drivers/display/display_nt35510.c

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright (c) 2024 Erik Andersson <erian747@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Based on the NT35510 driver provided by STMicroelectronics at
+ * https://github.com/STMicroelectronics/stm32-nt35510/blob/main/nt35510.c
+ */
+
+#define DT_DRV_COMPAT frida_nt35510
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/mipi_dsi.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+#include "display_nt35510.h"
+
+LOG_MODULE_REGISTER(nt35510, CONFIG_DISPLAY_LOG_LEVEL);
+
+/**
+ * @brief  Possible values of COLMOD parameter corresponding to used pixel formats
+ */
+#define NT35510_COLMOD_RGB565 0x55
+#define NT35510_COLMOD_RGB888 0x77
+
+/**
+ * @brief  NT35510_480X800 Timing parameters for Portrait orientation mode
+ */
+#define NT35510_480X800_HSYNC ((uint16_t)2)  /* Horizontal synchronization */
+#define NT35510_480X800_HBP   ((uint16_t)34) /* Horizontal back porch      */
+#define NT35510_480X800_HFP   ((uint16_t)34) /* Horizontal front porch     */
+
+#define NT35510_480X800_VSYNC ((uint16_t)120) /* Vertical synchronization   */
+#define NT35510_480X800_VBP   ((uint16_t)150) /* Vertical back porch        */
+#define NT35510_480X800_VFP   ((uint16_t)150) /* Vertical front porch       */
+
+/**
+ * @brief  NT35510_800X480 Timing parameters for Landscape orientation mode
+ *         Same values as for Portrait mode in fact.
+ */
+#define NT35510_800X480_HSYNC NT35510_480X800_VSYNC /* Horizontal synchronization */
+#define NT35510_800X480_HBP   NT35510_480X800_VBP   /* Horizontal back porch      */
+#define NT35510_800X480_HFP   NT35510_480X800_VFP   /* Horizontal front porch     */
+#define NT35510_800X480_VSYNC NT35510_480X800_HSYNC /* Vertical synchronization   */
+#define NT35510_800X480_VBP   NT35510_480X800_HBP   /* Vertical back porch        */
+#define NT35510_800X480_VFP   NT35510_480X800_HFP   /* Vertical front porch       */
+
+struct nt35510_config {
+	const struct device *mipi_dsi;
+	const struct gpio_dt_spec reset;
+	const struct gpio_dt_spec backlight;
+	uint8_t data_lanes;
+	uint16_t width;
+	uint16_t height;
+	uint8_t channel;
+	uint16_t rotation;
+};
+
+struct nt35510_data {
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+	uint16_t xres;
+	uint16_t yres;
+};
+
+struct nt35510_init_cmd {
+	uint8_t reg;
+	uint8_t cmd_len;
+	uint8_t cmd[5];
```

Comment:
Don't know if you care to do this, but you can probably save flash space by making this `cmd` field a pointer, and then declaring the init command arrays separately (like so):

```
static const uint8_t lv2_cmds[] = {0x55, 0xaa, 0x52, 0x08, 0x01};

static const struct nt35510_init_cmd init_cmds[] = {
	/* LV2:  Page 1 enable */
	{.reg = 0xf0, .cmd_len = sizeof(lv2_cmds), .cmd = lv2_cmds},
```

Reply:
We could do something like this, my concern is implementing parsing code for that case seems unnecessarily complicated. Truthfully I'm not very worried about the extra few bytes used by this approach- the parts this display will be targeting generally have a lot of flash space.

If you want to implement this method or the one I suggested above, feel free to- but I'm also fine with the implementation as is

Reply:
Yes that's true, a few bytes extra here would not be critical in a system with a high resolution display :) 
Then i leave it as it is right now